### PR TITLE
Add workflow support

### PR DESCRIPTION
**Pull request by BenjaminGill-Metaswitch**
This seems to compile, but I haven't managed to actually test this properly...

Should resolve [JENKINS-42629](https://issues.jenkins-ci.org/browse/JENKINS-42629)

~- [ ] Work out whether testable~
~- [ ] Write up necessary configuration in pipelines (declarative and legacy)~

### CODE REVIEW

**Comment by Daniel15** [comment]
Thanks for the pull request. I'd like to get confirmation that it actually works before merging the pull request 😃 

Some unit tests would be nice but I know very little about Java and am not sure how to write unit tests for Jenkins plugins :(

**Comment by BenjaminGill-Metaswitch** [comment]
OK. I've now installed the new package on a production system, and it appears to work (i.e. it can now raise issues within a pipeline where before it couldn't).

I don't really understand the UT framework here, but will have a go at seeing if I can test this...

**Comment by BenjaminGill-Metaswitch** [comment]
Right. I've been using this for the past 10 days in production without significant problems. I've still not worked out how the UTs work for this plugin, though. 

I think the correct place for documentation will be https://wiki.jenkins-ci.org/display/JENKINS/GitHub+Issues+Plugin, once a release appears with this feature:

As an example (note that the GitHub project must be configured in the main Jenkins config):

```groovy
script {
    properties([[$class: 'GithubProjectProperty',
                projectUrlStr: '<GitHub repo URL>']])
}

...

step([$class: 'GitHubIssueNotifier',
      issueAppend: true,
      issueLabel: '',
      issueTitle: '$JOB_NAME $BUILD_DISPLAY_NAME failed'])
```

Is this worth merging now?

**Comment by BenjaminGill-Metaswitch** [comment]
Is this able to be merged?

**Comment by Daniel15** [comment]
Sorry, I'll take a look soon. Thanks for your contribution!

File: src/main/java/org/jenkinsci/plugins/githubissues/GitHubIssueNotifier.java [hunk 1]
```diff
@@ -181,15 +182,15 @@ public void perform(
                 logger.format("GitHub Issue Notifier: Build has started failing, filed GitHub issue #%s%n", issue.getNumber());
                 run.addAction(new GitHubIssueAction(issue, GitHubIssueAction.TransitionAction.OPEN));
             }
-        } else if (result == Result.SUCCESS && issue != null && issue.getState() == GHIssueState.OPEN) {
+        } else if ((result == Result.SUCCESS || result == null) && issue != null && issue.getState() == GHIssueState.OPEN) {
```
Comment (Daniel15) [inline src/main/java/org/jenkinsci/plugins/githubissues/GitHubIssueNotifier.java]: Why do you need to add `result == null`?

Comment (BenjaminGill-Metaswitch) [reply src/main/java/org/jenkinsci/plugins/githubissues/GitHubIssueNotifier.java]: In declarative pipelines, the result will be `null` unless explicitly set. The common pattern is to set the failure state within a `catch` block, so we can treat `null` as `SUCCESS`. I've added an explanatory comment.

Comment (Daniel15) [reply src/main/java/org/jenkinsci/plugins/githubissues/GitHubIssueNotifier.java]: Thank you for the explanation :)

**Comment by Daniel15** [comment]
Published as v1.2.3. Thanks for your contribution!

**Pull request merged**